FWCore/Framework/interface/EventSetupRecordImpl.h

```diff
@@ -167,7 +173,7 @@ namespace edm {
                              std::shared_ptr<ESHandleExceptionFactory>& whyFailedFactory) const {
         DataKey const* dataKey = nullptr;
         if (iResolverIndex.value() == std::numeric_limits<int>::max()) {
-          whyFailedFactory = makeESHandleExceptionFactory([=] {
+          whyFailedFactory = makeESHandleExceptionFactory([=, this] {
```

Comment:
This file should be easy to migrate without suppressing the warning
```suggestion
          whyFailedFactory = makeESHandleExceptionFactory([this] {
```